s3-import-commons/src/test/java/no/unit/nva/publication/s3imports/DoiLinkUpdateEventEmitterTest.java

```diff
@@ -0,0 +1,120 @@
+        InputStream inputStream = toInputStream(eventReference);
+        handler.handleRequest(inputStream, outputStream, context);
+        var updatedPublications = getUpdatedPublications(publicationsToModify);
+        updatedPublications.forEach(publication -> publication.setModifiedDate(null));
+
+        assertThat(expectedPublications, (Every.everyItem(hasProperty("doi", is(equalTo(null))))));
+        assertThat(expectedPublications, containsInAnyOrder(updatedPublications.toArray()));
+    }
+
+    private static String getPublicationInCSVFormat(List<Publication> publications) {
+        return publications.stream().map(p -> p.getIdentifier().toString() + "\n").collect(
+            Collectors.toList()).toString().replaceAll("[\\(\\)\\[\\]\\{\\}\\,]", "");
+    }
+
+    private static Publication moveDoiToNewField(Publication publication) {
+        publication.getEntityDescription().getReference().setDoi(publication.getDoi());
+        publication.setDoi(null);
+        return publication;
+    }
+
+    private List<Publication> getUpdatedPublications(List<Publication> publicationsToModify) {
+        return publicationsToModify.stream()
+                   .map(Publication::getIdentifier)
+                   .map(identifier -> {
+                       try {
+                           return resourceService.getPublicationByIdentifier(identifier);
+                       } catch (nva.commons.apigateway.exceptions.NotFoundException e) {
+                           throw new RuntimeException(e);
+                       }
+                   }).map(p -> p.copy().build())
+                   .collect(Collectors.toList());
+    }
+
+    private Publication persist(Publication publication) {
+        UserInstance userInstance = UserInstance.fromPublication(publication);
+        return Resource.fromPublication(publication).persistNew(resourceService, userInstance);
+    }
+
+    private List<Publication> createPublicationWithDoiValueInLinkField(List<Publication> publicationsToModify) {
+        return publicationsToModify.stream()
+                   .map(DoiLinkUpdateEventEmitterTest::moveDoiToNewField)
+                   .collect(Collectors.toList());
+    }
+
+    private InputStream toInputStream(EventReference request) {
+        return attempt(() -> s3ImportsMapper.writeValueAsString(request))
+                   .map(IoUtils::stringToStream)
+                   .orElseThrow();
+    }
+
+    private List<Publication> createPublications() {
+        var publications = IntStream.range(0, 10)
+                               .boxed()
+                               .map(i -> persist(randomPublication()))
+                               .collect(Collectors.toList());
+        s3Client.putObject(
+            PutObjectRequest.builder().bucket("nve-doi-to-update-bucket").key(DEFAULT_FILE_NAME).build(),
+            RequestBody.fromString(getPublicationInCSVFormat(publications)));
+        return publications;
+    }
+}
```

Comment:
- Again, I moved things around again for readability.
- Not sure what the regex (☠️) in the csv conversion was about, I replaced it with something less involved
- A few vars and formatting things too.
- Look over the code and tell me what you think :)

```
class DoiLinkUpdateEventEmitterTest extends ResourcesLocalTest {

    public static final String DEFAULT_FILE_NAME = "results.csv";
    private static final Context context = mock(Context.class);
    private FakeS3Client s3Client;
    private ResourceService resourceService;
    private DoiLinkUpdateEventEmitter handler;

    @BeforeEach
    public void init() {
        super.init();
        this.s3Client = new FakeS3Client();
        this.resourceService = new ResourceService(client, Clock.systemDefaultZone());
        this.handler = new DoiLinkUpdateEventEmitter(s3Client, resourceService);
    }

    @Test
    void shouldMoveDoiValueToLinkField() throws IOException, NotFoundException {
        var publicationsToModify = createPublications();
        var eventReference = createEventReference(publicationsToModify);

        var inputStream = toInputStream(eventReference);
        handler.handleRequest(inputStream, new ByteArrayOutputStream(), context);
        var updatedPublications = fetchUpdatedPublications(publicationsToModify);

        var expectedPublications = createPublicationWithDoiValueInLinkField(publicationsToModify);

        assertThat(expectedPublications, (Every.everyItem(hasProperty("doi", is(equalTo(null))))));
        assertThat(expectedPublications, containsInAnyOrder(updatedPublications.toArray()));
    }

    private EventReference createEventReference(List<Publication> publications) {
        s3Client.putObject(
            PutObjectRequest.builder().bucket("nve-doi-to-update-bucket").key(DEFAULT_FILE_NAME).build(),
            RequestBody.fromString(getPublicationInCSVFormat(publications)));
        return new EventReference(null, null, URI.create("s3://nve-doi-to-update-bucket"
                                                         + "/results.csv"));
    }

    private static String getPublicationInCSVFormat(List<Publication> publications) {
        return publications.stream()
                   .map(Publication::getIdentifier)
                   .map(SortableIdentifier::toString)
                   .collect(Collectors.joining("\n"));
    }

    private static Publication moveDoiToNewField(Publication publication) {
        publication.getEntityDescription().getReference().setDoi(publication.getDoi());
        publication.setDoi(null);
        return publication;
    }

    private List<Publication> fetchUpdatedPublications(List<Publication> publicationsToModify) {
        var publications = publicationsToModify.stream()
                               .map(Publication::getIdentifier)
                               .map(identifier -> attempt(
                                   () -> resourceService.getPublicationByIdentifier(identifier)).orElseThrow())
                               .collect(Collectors.toList());
        publications.forEach(publication -> publication.setModifiedDate(null));
        return publications;
    }

    private Publication persist(Publication publication) {
        var userInstance = UserInstance.fromPublication(publication);
        return Resource.fromPublication(publication).persistNew(resourceService, userInstance);
    }

    private List<Publication> createPublicationWithDoiValueInLinkField(List<Publication> publicationsToModify) {
        var publications = publicationsToModify.stream()
                               .map(DoiLinkUpdateEventEmitterTest::moveDoiToNewField)
                               .collect(Collectors.toList());
        publications.forEach(i -> i.setModifiedDate(null));
        return publications;
    }

    private InputStream toInputStream(EventReference request) {
        return attempt(() -> s3ImportsMapper.writeValueAsString(request))
                   .map(IoUtils::stringToStream)
                   .orElseThrow();
    }

    private List<Publication> createPublications() {
        return IntStream.range(0, 10)
                   .boxed()
                   .map(i -> persist(randomPublication()))
                   .collect(Collectors.toList());
    }
}
```

Reply:
Regex was for removing all necessary brackets from the string when converting List to String :) 
Test is more readable and you have removed all not unnecessary methods